include/mqt-core/fomac/FoMaC.hpp

```diff
@@ -141,30 +143,64 @@ concept maybe_optional_value_or_string_or_vector =
     value_or_string_or_vector<remove_optional_t<T>>;
 
 /// @returns the string representation of the given QDMI_STATUS.
-auto toString(QDMI_STATUS result) -> std::string;
+auto toString(const QDMI_STATUS result) -> std::string;
 
 /// @returns the string representation of the given QDMI_Site_Property.
-auto toString(QDMI_Site_Property prop) -> std::string;
+auto toString(const QDMI_Site_Property prop) -> std::string;
 
 /// @returns the string representation of the given QDMI_Operation_Property.
-auto toString(QDMI_Operation_Property prop) -> std::string;
+auto toString(const QDMI_Operation_Property prop) -> std::string;
 
 /// @returns the string representation of the given QDMI_Device_Property.
-auto toString(QDMI_Device_Property prop) -> std::string;
+auto toString(const QDMI_Device_Property prop) -> std::string;
 
 /// @returns the string representation of the given QDMI_Session_Property.
-constexpr auto toString(QDMI_Session_Property prop) -> std::string {
+constexpr auto toString(const QDMI_Session_Property prop) -> std::string {
   if (prop == QDMI_SESSION_PROPERTY_DEVICES) {
     return "QDMI_SESSION_PROPERTY_DEVICES";
   }
   return "QDMI_SESSION_PROPERTY_UNKNOWN";
 }
 
+/**
+ * @brief Session parameters for authentication and configuration.
+ * @details These parameters must be set before the session is initialized
+ * (i.e., before the first call to getDevices()).
+ * @see QDMI_SESSION_PARAMETER_T
+ */
+enum class SessionParameter {
+  /// Authentication token
+  TOKEN = QDMI_SESSION_PARAMETER_TOKEN,
+  /// Path to authentication file
+  AUTHFILE = QDMI_SESSION_PARAMETER_AUTHFILE,
+  /// URL to authentication server
+  AUTHURL = QDMI_SESSION_PARAMETER_AUTHURL,
+  /// Username for authentication
+  USERNAME = QDMI_SESSION_PARAMETER_USERNAME,
+  /// Password for authentication
+  PASSWORD = QDMI_SESSION_PARAMETER_PASSWORD,
+  /// Project ID for session
+  PROJECTID = QDMI_SESSION_PARAMETER_PROJECTID,
+  /// Custom parameter 1 (driver-defined)
+  CUSTOM1 = QDMI_SESSION_PARAMETER_CUSTOM1,
+  /// Custom parameter 2 (driver-defined)
+  CUSTOM2 = QDMI_SESSION_PARAMETER_CUSTOM2,
+  /// Custom parameter 3 (driver-defined)
+  CUSTOM3 = QDMI_SESSION_PARAMETER_CUSTOM3,
+  /// Custom parameter 4 (driver-defined)
+  CUSTOM4 = QDMI_SESSION_PARAMETER_CUSTOM4,
+  /// Custom parameter 5 (driver-defined)
+  CUSTOM5 = QDMI_SESSION_PARAMETER_CUSTOM5
+};
+
```

Comment:
Why do we need to re-expose this enum when there already is a perfectly fine enum? Doesn't this create an additional level of abstraction that we probably do not want?